spec/system/authentication/conditional_registration_spec.rb

```diff
@@ -0,0 +1,108 @@
+require "rails_helper"
+
+# This test file can be removed once we have a longterm solution for
+# ForemWebView contexts when Apple Auth isn't enabled
+RSpec.describe "Conditional registration (ForemWebView)", type: :system do
+  let(:all_providers) { Authentication::Providers.available }
+  let(:all_providers_except_apple) { Authentication::Providers.available - [:apple] }
+  let(:mobile_browser_ua) { "Mozilla/5.0 (iPhone) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148" }
+  let(:foremwebview_ua) do
+    "Mozilla/5.0 (iPhone) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 ForemWebView/1.0"
+  end
+  let(:flow_b_fallback_text) do
+    "Unfortunately, we do not support creating new accounts right now on our "\
+      "mobile app. If you want create a new account to join "\
+      "#{Settings::Community.community_name}, please do that on the web."
+  end
+
+  before do
+    Flipper.enable(:apple_auth)
```

Comment:
tiny and non blocking, you can use the wrapper class `FeatureFlag`:

```suggestion
    FeatureFlag.enable(:apple_auth)
```

even better if you mock that, so that `apple_auth` won't be enabled after this test is finished:

```suggestion
    allow(FeatureFlag).to receive(:enabled?).with(:apple_auth).and_return(true)
```

Reply:
Thanks for the suggestion! I keep forgetting about `FeatureFlag` because of the gem's name is right there in mind, also couldn't agree more that using the mock is indeed a better approach 👍